lib/chef/provider/ifconfig/debian.rb

```diff
@@ -32,25 +32,43 @@ class Debian < Chef::Provider::Ifconfig
         def initialize(new_resource, run_context)
           super(new_resource, run_context)
           @config_template = %{
-<% if new_resource.device %>
-<% if new_resource.onboot == "yes" %>auto <%= new_resource.device %><% end %>
+<% if new_resource.device -%>
```

Comment:
If the whole thing depends on `new_resource.device`, that conditional could be moved out of the ERB string and just set `@config_template = ""` or something.